keyboards/bastardkb/dilemma/4x6_4/4x6_4.c

```diff
@@ -57,3 +57,59 @@ bool encoder_update_kb(uint8_t index, bool clockwise) {
     return true;
 }
 #endif // ENCODER_ENABLE
+
+#ifdef RGB_MATRIX_ENABLE
+// Layer state indicator
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
+    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) { return false; }
+    if (host_keyboard_led_state().caps_lock) {
+        for (int i = led_min; i <= led_max; i++) {
+            if (HAS_FLAGS(g_led_config.flags[i], LED_FLAG_MODIFIER)) {
+                rgb_matrix_set_color(i, MIN(rgb_matrix_get_val() + 76, 255), 0x00, 0x00);
+            }
+        }
+    }
+
+    uint8_t layer = get_highest_layer(layer_state);
```

Comment:
It seems weird to me that _layer_ indicators are implemented in the keyboard-specific code, when they should be very much a keymap implementation details, but if that's what make QMK people happy…

Reply:
It's definitely a QMK requirement, but from another perspective this has the benefit of providing a default RGB indicators behaviour which can be relatively simply overriden in keymap level for all users without needing to duplicate code